feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts

```diff
@@ -80,10 +125,46 @@ export class ConfiguratorAddToCartButtonComponent implements OnInit, OnDestroy {
     protected orderHistoryFacade: OrderHistoryFacade,
     protected commonConfiguratorUtilsService: CommonConfiguratorUtilsService,
     protected configUtils: ConfiguratorStorefrontUtilsService,
-    protected intersectionService: IntersectionService
+    protected intersectionService: IntersectionService,
+    @Optional()
+    protected configuratorQuantityService?: ConfiguratorQuantityService
   ) {}
+
   ngOnInit(): void {
     this.makeAddToCartButtonSticky();
+
+    if (this.configuratorQuantityService) {
+      this.configuratorQuantityService
+        .getQuantity()
+        .pipe(take(1))
+        .subscribe((quantity) => {
+          this.quantityControl.setValue(quantity);
+        });
+    }
+
+    this.subscription.add(
+      this.quantityControl.valueChanges
+        .pipe(distinctUntilChanged())
+        .subscribe(() =>
+          this.configuratorQuantityService?.setQuantity(
+            this.quantityControl.value
+          )
+        )
+    );
+  }
+
+  /**
+   * Retrieves a quantity for a cart entry.
+   *
+   * @param {string} entryNumber - Entry number
+   * @returns {number} - Quantity
+   */
+  getQuantity(entryNumber: string): Observable<number | undefined> {
```

Comment:
Change method name to indicate that it's about cart entries only
```suggestion
  getCartEntryQuantity(entryNumber: string): Observable<number | undefined> {
```

Reply:
Agree!

Reply:
Done!